src/plugins/intel_cpu/src/nodes/mkldnn_input_node.cpp

```diff
@@ -254,7 +254,17 @@ void MKLDNNInputNode::cloneBlobIfRequired() {
 
     auto cloneBlob = [&, this] () {
         MKLDNNMemory memory{ getEngine() };
-        memory.Create(memDesc, constOp->get_data_ptr());
+
+        // CVS-74980
+        // MKLDNN/oneDNN always allocate 1byte for element type with bitWidth < 8 (u4,u1...)
+        // but ngraph Constant uses actual bitWidth for data storage allocation
+        // in that case we make a copy to avoid overflow
+        if (constOp->get_byte_size() >= memDesc.getCurrentMemSize()) {
+            memory.Create(memDesc, constOp->get_data_ptr());
+        } else {
+            memory.Create(memDesc);
+            memcpy(memory.GetPtr(), constOp->get_data_ptr(), constOp->get_byte_size());
```

Comment:
To be honest I can not fully understand the proposed approach. As we can see from the comment, we need to allocate one byte per each data value, and we do it calling `memory.Create(memDesc)` where required memory is allocated, then we just copy the binary blob to the allocated memory. But we do not left zero extend each value so that it occupies a byte, in other word we just move the same binary blob that we read before but into larger memory space. Then we call `SetData` where the data are copied/reordered, that means we do not use the `memory` further. Could you please elaborate where left zero extension to byte size happens for each value and why we have to allocate memory of size NuberOfValues in bytes while we store the binary blob of less size (since in IR every bit is used) there. Because if oneDNN reads the data assuming that each value occupies a byte, we can get into situation where the oneDNN reads invalid data from the spare space after the copied data blob.

Reply:
I'm 100% sure, but oneDNN didn't support bin data type and @dmitry-gorokhov added the support about a year ago, and due to oneDNN API's limitation, element of each data type should have a size in unit of byte (`types::data_type_size()`), so all memory of `bin` type actually allocate 8 times memory more than actually needed.

From the way `MKLDNNBinaryConvolutionNode::executeReference` accessing input memory 
and the way `MKLDNNFakeQuantizeNode::executeReference` accessing output memory, it's clearly a bit-packed format rather than left-zero extended format is being expected. so the only important data in `bin` type memory is the first 1/8, the other extra memory area seems to be not important. I'm not 100% sure of this point, so maybe @dmitry-gorokhov  can clarify this.

Ngraph allocates BIN type memory of correct size w/o extra memory paddings, and that triggers problem in cloneBlobIfRequired(), which wraps ngraph memory as MKLDNNMemory, when later on  this MKLDNNMemory is being copied/reordered, overflow(read) would happen and sometime a non-accessible region following it will be accessed and segment-fault happens.

Reply:
Uh-huh, reading a non allocated memory region is the problem here. I suppose due to vectorized copy. Now I understand. Thank you.